From 2e0d78265df5bfcc27585a7571542c7a618855cb Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Tue, 27 May 2025 12:54:26 +0300
Subject: [PATCH] [SW-228856] fit: use new fit_services infra

Context:
We unified some fit related code.
The main lego pieces are the newly created (via other commits):
fit_services.c/h located (in emb_common_mod).
The skeletton is ready and meanwhile it includes the use of
fit_foreach_property  (since there were duplications of code!)
So from now, please, anyone doing an infrastructure change related to
the fit files should carefully double-check.
Since any unfied code should be located In fit_services.c/h in a
similar way as fit_foreach_property has been.

For this specific commit:
In this current commit, we do the changes needed to the use
unified fit_foreach_property in Preboot.
Both Makefiles and c and h files are adapted as needed.

Change-Id: I164e67613a940bb77f1a37dd86a74b446fcd604d
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d980602..9b85731 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -115,6 +115,7 @@
 zephyr_library_sources(emb_common_mod/lib/common/lib_common.c)
 zephyr_library_sources(emb_common_mod/lib/common/boot_desc.c)
 zephyr_library_sources(emb_common_mod/lib/common/pci.c)
+zephyr_library_sources(emb_common_mod/lib/common/fit_services.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_rsa.c)
 zephyr_library_sources(emb_common_mod/lib/preboot/fit_regions.c)
diff --git a/include/preboot.h b/include/preboot.h
index 297f002..fff8614 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -35,6 +35,7 @@
 #include <stdbool.h>
 #include <types.h>
 #include "auto_generated.h"
+#include "fit_services.h"
 
 #define VERSION_MAX_LEN		128
 
@@ -64,13 +65,6 @@
 enum preboot_vers {
 	PREBOOT_BTL_VER,
 	PREBOOT_EFUSE_VER,
-};
-
-enum fit_sec_failure {
-	FIT_SEC_NONE = 0,
-	FIT_SEC_SVN_INVALID,
-	FIT_SEC_KEY_REVOKED,
-	FIT_SEC_SIGN_FAILED,
 };
 
 struct hl_nic_en {
diff --git a/lib/module/module_lkd.c b/lib/module/module_lkd.c
index 5479d50..134baa0 100644
--- a/lib/module/module_lkd.c
+++ b/lib/module/module_lkd.c
@@ -25,6 +25,7 @@
 
 #include "cmd_module.h"
 #include "fit.h"
+#include "fit_services.h"
 #include "security.h"
 #include <watchdog_timer.h>
 
diff --git a/lib/security/security.c b/lib/security/security.c
index e03917d..d66527d 100644
--- a/lib/security/security.c
+++ b/lib/security/security.c
@@ -23,6 +23,7 @@
 #include <hl_fit_if.h>
 #include <hl_boot_info.h>
 #include <hl_boot_desc.h>
+#include "fit_services.h"
 
 #define LOG_LEVEL CONFIG_LOG_DEFAULT_LEVEL
 #include <zephyr/logging/log.h>
